executor/src/witgen/query_processor.rs

```diff
@@ -107,41 +109,59 @@ where
                 *op,
                 self.evaluate_expression(right, rows)?,
             )),
-            Expression::UnaryOperation(op, expr) => Ok(evaluate_unary_operation(
-                *op,
-                self.evaluate_expression(expr, rows)?,
-            )),
-            Expression::Reference(Reference::LocalVar(i, _name)) => {
-                assert!(*i == 0);
-                Ok(rows.current_row_index.into())
-            }
-            Expression::Reference(Reference::Poly(poly)) => {
-                if poly.index.is_none() {
-                    let poly_id = poly.poly_id.unwrap();
-                    match poly_id.ptype {
-                        PolynomialType::Committed | PolynomialType::Intermediate => {
-                            let poly_ref = AlgebraicReference {
-                                name: poly.name.clone(),
-                                poly_id,
-                                index: poly.index,
-                                next: false,
-                            };
-                            Ok(rows
-                                .get_value(&poly_ref)
-                                .ok_or(IncompleteCause::DataNotYetAvailable)?)
-                        }
-                        PolynomialType::Constant => Ok(self.fixed_data.fixed_cols[&poly_id].values
-                            [rows.current_row_index as usize]),
-                    }
+            Expression::UnaryOperation(op, expr) => {
+                // Special case to allow `x'` for now on witness columns.
+                // TODO: Check if we actually need this or should replace it by `x(i + 1)`
+                if let Expression::Reference(Reference::Poly(poly)) = expr.as_ref() {
```

Comment:
and if `op == UnaryOperation::Next`, right?

Reply:
haha yes! :)